internal/module/services/gateway/gateway.go

```diff
@@ -606,3 +614,140 @@ func (gw *TerrariumGrpcGateway) RetrieveModuleDependenciesWithClient(request *te
 		}
 	}
 }
+
+// Publish a new release with Release service
+func (gw *TerrariumGrpcGateway) Publish(ctx context.Context, request *releasePkg.PublishRequest) (*releasePkg.PublishResponse, error) {
+	conn, err := services.CreateGRPCConnection(releaseSvc.ReleaseServiceEndpoint)
+
+	if err != nil {
+		return nil, ConnectToReleaseError
+	}
+
+	defer conn.Close()
+
+	client := releasePkg.NewReleasePublisherClient(conn)
+
+	return gw.PublishWithClient(ctx, request, client)
+}
+
+// PublishWithClient calls Publish on Release client
+func (gw *TerrariumGrpcGateway) PublishWithClient(ctx context.Context, request *releasePkg.PublishRequest, client releasePkg.ReleasePublisherClient) (*releasePkg.PublishResponse, error) {
+	span := trace.SpanFromContext(ctx)
+	span.SetAttributes(
+		attribute.String("release.name", request.GetName()),
+		attribute.String("release.version", request.GetVersion()),
+		attribute.String("release.type", request.GetType()),
+		attribute.String("release.organization", request.GetOrganization()),
+	)
+	defer span.End()
```

Comment:
@aflahahamed  / @vmilovanovicc - you going to be annoyed with me but I've realised that because you're not creating new spans only retrieving the already created span we don't need the `defer span.End()`, sorry!

If we had been creating new spans here with `tracer.Start()` we would need to ensure that the span was End'ed.

This applies to all the places in this file where I told you to add them...